### PR TITLE
Rating: add tooltip for the items

### LINKED ISSUE #5292: Rating: add tooltip for the items
This will make it easy for the user to indicate the right score

**Example**

image

### PR DESCRIPTION
**Pull request by stsrki**
Closes #5292

### CODE REVIEW

File: Source/Blazorise/Models/RatingItemTooltipInfo.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+﻿namespace Blazorise;
+
+/// <summary>
+/// Defines the tooltip information for the <see cref="Rating"/> component.
+/// </summary>
+public class RatingItemTooltipInfo
```
Comment (David-Moreira) [inline Source/Blazorise/Models/RatingItemTooltipInfo.cs:6]: What about **RatingItemTooltip** You are defining how the actual tooltip gets constructed not just the "info".

File: Source/Blazorise/Models/RatingItemTooltipInfo.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+﻿namespace Blazorise;
+
+/// <summary>
+/// Defines the tooltip information for the <see cref="Rating"/> component.
```
Comment (David-Moreira) [inline Source/Blazorise/Models/RatingItemTooltipInfo.cs:4]: Is it for the `Rating` component or the `RatingItem` component?

File: Source/Blazorise/Components/Rating/RatingItem.razor [hunk 3]
```diff
@@ -1,14 +1,33 @@
 ﻿@namespace Blazorise
 @inherits BaseComponent
-@if ( Rating.ReadOnly )
+@{
+    var tooltipInfo = Rating?.GetItemTooltipInfo?.Invoke( Value );
+}
+
+@if ( tooltipInfo is null || string.IsNullOrEmpty( tooltipInfo.Text ) )
```
Comment (David-Moreira) [inline Source/Blazorise/Components/Rating/RatingItem.razor:7]: can probably do `string.IsNullOrEmpty( tooltipInfo?.Text )`. Not a bother.

File: Source/Blazorise/Components/Rating/Rating.razor.cs [hunk 4]
```diff
@@ -225,5 +226,10 @@ public int SelectedValue
     /// </summary>
     [Parameter] public EventCallback<int?> HoveredValueChanged { get; set; }
 
+    /// <summary>
+    /// Expression that allows the rating item to have additional user-friendly information.
+    /// </summary>
+    [Parameter] public Func<int?, RatingItemTooltipInfo> GetItemTooltipInfo { get; set; }
```
Comment (David-Moreira) [inline Source/Blazorise/Components/Rating/Rating.razor.cs:232]: Maybe just GetItemTooltip?

Comment (David-Moreira) [reply Source/Blazorise/Components/Rating/Rating.razor.cs:232]: Is there a Rating Item that's represented by null? That's why we have nullable int? Would we have an actual tooltip render when the value is null?

Comment (David-Moreira) [reply Source/Blazorise/Components/Rating/Rating.razor.cs:232]: Also is it ok to use `Get` prefix for `Func` from a coding style perspective? I'm not sure we use that approach, but we do seem pretty lax about it across the code base. 
Looking at some of them, could it perhaps be `ItemTooltipCreator`?

File: Source/Blazorise/Models/RatingItemTooltipInfo.cs (hunk 1 above)
Comment (stsrki) [reply Source/Blazorise/Models/RatingItemTooltipInfo.cs]: Renamed it to `RatingTooltip`. We already don't actively expose `RatingItem`, only the `Rating` component. So this makes the most sense.

File: Source/Blazorise/Components/Rating/Rating.razor.cs (hunk 4 above)
Comment (stsrki) [reply Source/Blazorise/Components/Rating/Rating.razor.cs]: I don't think `Creator` makes sense. This is just a delegate. So from a delegate perspective the `Get` prefix makes the most sense. And we already have that naming in other places, TreeView, for example.

File: Source/Blazorise/Components/Rating/Rating.razor.cs [hunk 5]
```diff
@@ -225,5 +226,13 @@ public int SelectedValue
     /// </summary>
     [Parameter] public EventCallback<int?> HoveredValueChanged { get; set; }
 
+    /// <summary>
+    /// Expression that provides the tooltip for the rating item to have additional user-friendly information.
+    /// </summary>
+    /// <remarks>
+    /// The expression receives the rating item <Code>Value</Code> to determine the result.
+    /// </remarks>
+    [Parameter] public Func<int?, RatingTooltip> GetTooltip { get; set; }
```
Comment (David-Moreira) [inline Source/Blazorise/Components/Rating/Rating.razor.cs:235]: Is there a Rating Item that's represented by null? That's why we have nullable int? Would we have an actual tooltip render when the value is null? What happens?

**Review by David-Moreira — APPROVED** [review]
LGTM

File: Source/Blazorise/Components/Rating/Rating.razor.cs (hunk 5 above)
Comment (David-Moreira) [reply Source/Blazorise/Components/Rating/Rating.razor.cs]: Made*? :P

**Pull request merged**